README.md

```diff
@@ -237,6 +237,180 @@ You can do the same for Kafka Streams and Kafka, by using the `kstreams` and `ka
 
 Lastly, create a Makefile in the `code` directory to invoke the harness runner and check any outputs that it produces. Then modify the `.semaphore/semaphore.yml` file to invoke that Makefile. This will make sure your tutorial gets checked by the CI system.
 
+## Working with the harness file
+
+Given the test harness is the `heart` of a tutorial, it will be helpful to describe in detail how to work with a `kafka|ksql|kstreams.yml` file.  It should be noted the harness file is in the [YAML file formt](https://en.wikipedia.org/wiki/YAML), so formatting properly is essential.  The harness files generates the structure of the rendered tutorial and also validates any output of tutorial steps against expected values.
+
+This section is not meant to be an exhaustive decription of the harness file.  New tutorial authors should not need to create a harness file from scratch, using either the `tools/gen_project.sh` or  `tools/clone.sh` script will provide a usable harness file.  This section should provide enough guidance to add, update, or remove sections as needed.
+
+#### TLDR;
+
+To run a tutorial programatically do the following stepes. Note these instructions assume you've already checked out the kafka-tutorials repo).
+
+1. cd into `_includes/tutorials/<tutorial name>/<type>/code` directory where type is one of `ksql | kstreams | kafka`.
+2. run the `make` command.
```

Comment:
I tried to run this myself and got:

```
“The ‘harness-runner==0.0.1’ distribution was not found and is required by the application”
```

1. How can I resolve this?
2. Consider whether this PR should add the explanation to resolve this error message

Reply:
It would be good to have a dedicated `prerequisites` section just for being able to test locally. Right now there is ruby, bundle, etc.. required, but I'm not sure this is needed just to run the harness-runner ? 
Ideally a docker version of the harness-runner would be great so that users don't have to install python3 and pip3

Reply:
@bbejeck as a reference example: 

https://github.com/confluentinc/examples/blob/5.5.0-post/clients/cloud/python/README.md#example-3-run-all-the-above-in-docker
https://github.com/confluentinc/examples/blob/5.5.0-post/clients/cloud/python/requirements.txt
https://github.com/confluentinc/examples/blob/5.5.0-post/clients/cloud/python/Dockerfile

I think you were thinking of addressing this in a separate PR.  But if you help me debug the harness runner error, I'm happy to build the Docker stuff.

Reply:
Dockerfile:

```
FROM python:3.7-slim
RUN pip3 install pyyaml
```

Build and execute:

```
docker build -t runner . ; docker run -v ${PWD}/harness_runner:/harness_runner/ -it --rm runner bash -c 'cd /harness_runner/ && pip3 install -e .'
```

And now make works:

```
(cd _includes/tutorials/transforming/kstreams/code/ && make)
```